experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,299 @@
+
+        for i in degree_list
+            if i^(sigma * x^-1 * y^-1) == i^(y*x*(x*y)^-1)
+                b = b + 1
+            end
+        end
+
+        # TODO can n_inv yields floating point errors?
+        if n_inv * (g - 1 - b * 0.5) + 1 == 0
+            return true
+        end
+    end
+
+    return false
+end
+
+function cylinder_structure(o::Origami)
+    # TODO returns Vector{Any}, maybe cast to Integer?
+    gap_obj = GAP.Globals.CylinderStructure(GapObj(o))
+    return GAP.gap_to_julia(gap_obj)
+end
+
+function veech_group_and_orbit(o::Origami)
+    return GAP.Globals.VeechGroupAndOrbit(GapObj(o))
+end
+
+function veech_group_is_even(o::Origami)
+    return GAP.Globals.VeechGroupIsEven(GapObj(o))
+end
+
+function are_equivalent(o1::Origami, o2::Origami)
+    return GAP.Globals.OrigamisEquivalent(GapObj(o1), GapObj(o2))
+end
+
+# TODO why is this not in canonical.jl?
+function normalform_conjugators(o::Origami)
+    x = horizontal_perm(o)
+    y = vertical_perm(o)
+    n = degree(o)
+    G = []
+
+    # Starting from each of the vertices found above, do a breadth-first search
+	# and list the vertices in the order they appear.
+	# This defines a permutation l with which we conjugate x and y.
+	# From the resulting list of pairs of permutations (all of which are by
+	# definition simultaneously conjugated to (x,y)) we choose the
+	# lexicographically smallest one as the canonical form.
+    for i in 1:n
+        L = fill(0, n)
+        seen = fill(false, n)
+        Q = [i]
+        seen[i] = true
+        numSeen = 1
+        L[i] = 1
+        while numSeen < n
+            v = popfirst!(Q)
+            wx = v^x
+            wy = v^y
+            if !seen[wx]
+                push!(Q, wx)
+                seen[wx] = true
+                numSeen = numSeen + 1
+                L[wx] = numSeen
+            end
+            if !seen[wy]
+                push!(Q, wy)
+                seen[wy] = true
+                numSeen = numSeen + 1
+                L[wy] = numSeen
+            end
+        end
+        push!(G, L)
+    end
+
+    return perm.(G)
+end
+
+function point_reflections(o::Origami)
+    if !veech_group_is_even(o)
+        throw("VeechGroup must contain -1")
+    end
+
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    o1 = origami(h^-1, v^-1)
+    h1 = horizontal_perm(o1)
+    v1 = vertical_perm(o1)
+    G = normalform_conjugators(o)
+    G1 = normalform_conjugators(o1)
+
+    f(i) = origami(i^-1 * h * i, i^-1 * v * i)
+    f1(i) = origami(i^-1 * h1 * i, i^-1 * v1 * i)
+
+    # origamis derived from the permutations above
+    O = f.(G)
+    # we need to calculate these to test find k s.t.
+    # sigma_i *origami *sigma_i^-1=delta_k(i)*origami_1*delta_k(i)^-1
+    O1 = f1.(G1)
+
+    # fitting the permuations together
+    result::Vector{PermGroupElem} = []
+
+    l = length(O)
+    for i in 1:l
+        index = findfirst(item -> item == O[i], O1)
+        push!(result, G[i] * (G1[index])^-1)
```

Comment:
"Dividing" two permutations can be done more efficiently than a multiplication plus an inversion
```suggestion
        push!(result, G[i] / G1[index])
```
Although the fastest results are for computing `a^-1*b` -- depending on what your function is supposed to return, exactly, maybe that'd be an option here?
```suggestion
        push!(result, G[i] \ G1[index])
```